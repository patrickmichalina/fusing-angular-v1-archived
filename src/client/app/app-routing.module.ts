import { NgModule } from '@angular/core'
import { RouterModule, Routes } from '@angular/router'

export function homeModule() {
  return import('./home/home.module').then(m => m.HomeModule)
}

export function aboutModule() {
  return import('./about/about.module').then(m => m.AboutModule)
}

export function demosModule() {
  return import('./demos/demos.module').then(m => m.DemosModule)
}

export function callbackModule() {
  return import('./callback/callback.module').then(m => m.CallbackModule)
}

export function logoutModule() {
  return import('./logout/logout.module').then(m => m.LogoutModule)
}

export function accountModule() {
  return import('./account/account.module').then(m => m.AccountModule)
}

export function readmeModule() {
  return import('./readme/readme.module').then(m => m.ReadmeModule)
}

export const routes: Routes = [
  { path: '', loadChildren: homeModule },
  { path: 'about', loadChildren: aboutModule },
  { path: 'demos', loadChildren: demosModule },
  { path: 'callback', loadChildren: callbackModule },
  { path: 'logout', loadChildren: logoutModule },
  { path: 'account', loadChildren: accountModule },
  { path: 'readme', loadChildren: readmeModule }
]

@NgModule({
  imports: [RouterModule.forRoot(routes, { initialNavigation: 'enabled' })],
  exports: [RouterModule]
})
export class AppRoutingModule {}
