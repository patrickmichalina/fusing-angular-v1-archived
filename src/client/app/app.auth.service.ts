import { AuthService } from './shared/auth/auth.service'
import { Injectable } from '@angular/core'

// You can customize your Auth configuration in this file!

export interface AppUser {
  readonly firstName?: string
  readonly name?: string
}

@Injectable()
export class AppAuthService extends AuthService<AppUser> {}
