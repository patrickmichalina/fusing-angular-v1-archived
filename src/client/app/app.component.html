<div>
  <pm-navbar (menuIconClick)="sidenav.toggle()"></pm-navbar>
  <div style="height: 52px;"></div>
  <mat-sidenav-container class="sidenav-container mat-typography" *ngIf="view$ | async as view">
    <mat-sidenav #sidenav class="sidenav">
      <mat-nav-list>
        <a routerLink mat-list-item title="Fusing Angular">
          <mat-icon pmSvg="logo"></mat-icon>
          <span>Fusing Angular</span>
        </a>
        <a routerLink="readme" mat-list-item title="I am a lazy module for the about page">
          <mat-icon matListIcon pmSvg="book_24"></mat-icon>
          <span>Readme</span>
        </a>
        <a routerLink="about" mat-list-item title="I am a lazy module for the about page">
          <mat-icon matListIcon pmSvg="functions_24"></mat-icon>
          <span>Performance</span>
        </a>
        <a routerLink="demos" mat-list-item title="I am a lazy module for the demos page">
          <mat-icon matListIcon pmSvg="important_devices_24"></mat-icon>
          <span>Isomorphic</span>
        </a>
        <a href="api-docs" mat-list-item title="View the rest API docs" pmExternalLink>
          <mat-icon matListIcon pmSvg="storage_24"></mat-icon>
          <span>API</span>
        </a>
        <div [ngSwitch]="view.loggedIn">
          <a *ngSwitchCase="0" mat-list-item (click)="login()">
            <mat-icon matListIcon pmSvg="home_24"></mat-icon>
            <span>Login</span>
          </a>
          <a *ngSwitchCase="1" mat-list-item (click)="logout()">
            <mat-icon matListIcon pmSvg="exit_to_app_24"></mat-icon>
            <span>Logout</span>
          </a>
        </div>
      </mat-nav-list>
    </mat-sidenav>
    <main id="route-wrapper">
      <router-outlet></router-outlet>
    </main>
  </mat-sidenav-container>
</div>