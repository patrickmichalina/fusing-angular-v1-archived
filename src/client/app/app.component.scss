@import '../styles/mixins';
:host {
  @include fill-vertical;
  .sidenav-container {
    @include fill-vertical;
  }
  #route-wrapper {
    @include fill-vertical;
    overflow-y: scroll;
    padding: 1em;
    @media screen and (max-width: 599px) {
      padding: 0.5em;
    }
  }
  .sidenav {
    width: 200px;
  }
  ::ng-deep mat-sidenav-content {
    @include fill-vertical;
  }
  ::ng-deep mat-sidenav-container {
    mat-sidenav {
      visibility: hidden;
    }
  }
  mat-nav-list {
    padding-top: 0;
  }
}
