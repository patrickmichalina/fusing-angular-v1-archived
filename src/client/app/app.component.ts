import { ChangeDetectionStrategy, Component } from '@angular/core'
import { Angulartics2GoogleAnalytics } from 'angulartics2/ga'
import { EnvironmentService } from './shared/services/environment.service'
import { InjectionService } from './shared/services/injection.service'
import { PlatformService } from './shared/services/platform.service'
import { AuthService } from './shared/services/auth.service'

@Component({
  selector: 'pm-app',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AppComponent {
  constructor(
    public analytics: Angulartics2GoogleAnalytics,
    es: EnvironmentService,
    is: InjectionService,
    auth: AuthService,
    ps: PlatformService
  ) {
    // tslint:disable-next-line:no-console
    console.log('logging environment: ', es.config)
    this.setBase(is, es.config.siteUrl)
    auth.user$.subscribe(user => {
      ps.isBrowser && user && analytics.setUsername(user.sub)
    })
    ps.isBrowser && auth.handleAuthentication()
    ps.isBrowser && auth.scheduleRenewal()
  }

  setBase(is: InjectionService, href = '/') {
    is
      .inject({
        inHead: true,
        element: 'base',
        attributes: {
          href
        }
      })
      .take(1)
      .subscribe()
  }
}
