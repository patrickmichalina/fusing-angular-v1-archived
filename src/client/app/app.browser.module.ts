import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
import {
  BrowserModule,
  BrowserTransferStateModule,
  TransferState
} from '@angular/platform-browser'
import { AppModule } from './app.module'
import { NgModule } from '@angular/core'
import { AppComponent } from './app.component'
import { ENV_CONFIG, ENV_CONFIG_TS_KEY, REQUEST_TS_KEY } from './app.config'
import { WINDOW } from './shared/services/utlities/window.service'
import { REQUEST } from '@nguniversal/express-engine/src/tokens'
import { ResponseService } from './shared/services/response.service'
import { LOGGER_CONFIG } from './shared/services/logging.service'
// import { ServiceWorkerModule, SwUpdate } from '@angular/service-worker'
// import { Observable } from 'rxjs/Observable'
// import 'hammerjs'

export function fuseBoxConfigFactory(ts: TransferState) {
  return ts.get(ENV_CONFIG_TS_KEY, {})
}

export function requestFactory(transferState: TransferState): any {
  return transferState.get<any>(REQUEST_TS_KEY, {})
}

@NgModule({
  bootstrap: [AppComponent],
  imports: [
    BrowserModule.withServerTransition({ appId: 'pm-app' }),
    BrowserTransferStateModule,
    BrowserAnimationsModule,
    AppModule
  ],
  providers: [
    { provide: WINDOW, useValue: window },
    {
      provide: ENV_CONFIG,
      useFactory: fuseBoxConfigFactory,
      deps: [TransferState]
    },
    {
      provide: REQUEST,
      useFactory: requestFactory,
      deps: [TransferState]
    },
    {
      provide: LOGGER_CONFIG,
      useValue: {
        name: 'Universal WebApp',
        type: 'client-side'
      }
    },
    ResponseService
  ]
})
export class AppBrowserModule {
  // tslint:disable:no-console
  // constructor(updates: SwUpdate) {
  //   Observable.interval(100000).subscribe(() => updates.checkForUpdate())
  //   updates.available.subscribe(event => {
  //     console.log('current version is', event.current)
  //     console.log('available version is', event.available)
  //   })
  //   updates.activated.subscribe(event => {
  //     console.log('old version was', event.previous)
  //     console.log('new version is', event.current)
  //   })
  // }
}
