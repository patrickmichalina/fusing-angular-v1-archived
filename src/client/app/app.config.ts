import { InjectionToken } from '@angular/core'
import { EnvConfig } from '../../../tools/config/app.config'
import { makeStateKey } from '@angular/platform-browser'

export const ENV_CONFIG = new InjectionToken<EnvConfig>('cfg.env')
export const ENV_CONFIG_TS_KEY = makeStateKey<EnvConfig>('cfg.env.ts')
export const REQUEST_TS_KEY = makeStateKey<IRequest>('cfg.req.ts')

export interface IRequest {
  readonly hostname?: string
  readonly entryReferer?: string
}
