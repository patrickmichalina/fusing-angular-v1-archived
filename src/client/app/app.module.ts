import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http'
import { AppComponent } from './app.component'
import { SharedModule } from './shared/shared.module'
import { AppRoutingModule } from './app-routing.module'
import { NotFoundModule } from './not-found/not-found.module'
import { BrowserModule, TransferState } from '@angular/platform-browser'
import { Angulartics2GoogleAnalytics } from 'angulartics2/ga'
import { Angulartics2Module } from 'angulartics2'
import { TransferHttpCacheModule } from '@nguniversal/common'
import { ErrorHandler, InjectionToken, NgModule } from '@angular/core'
import {
  GlobalErrorHandler,
  ROLLBAR_CONFIG,
  ROLLBAR_TS_KEY
} from './shared/services/error-handlers/global-error-handler.service'
import { EnvironmentService } from './shared/services/environment.service'
import {
  AUTH0_CLIENT,
  AUTH_ACCESS_TOKEN_EXPIRY_STORAGE_KEY,
  AUTH_ACCESS_TOKEN_STORAGE_KEY,
  AUTH_ID_TOKEN_STORAGE_KEY,
  AUTH_ROLES_KEY,
  AuthService,
  authZeroFactory,
  rolesKeyFactory
} from './shared/services/auth.service'
import * as Rollbar from 'rollbar'
import {
  AUTH_BEARER_HOSTS,
  HttpAuthInterceptor
} from './shared/services/http-interceptors/http-authorization-interceptor.service'
import { FirebaseModule } from './shared/firebase/firebase.module'

export const RXJS_DEFAULT_SCHEDULER = new InjectionToken<any>('cfg.rxjs.sch')

// export function cacheTagFactory(rs: ResponseService): any {
//   return (httpResponse: HttpResponse<any>, config: CacheTagConfig) => {
//     const cacheHeader = httpResponse.headers.get(config.headerKey)
//     cacheHeader && rs.appendHeader(config.headerKey, cacheHeader)
//   }
// }

export function rollbarFactory(ts: TransferState, es: EnvironmentService) {
  const accessToken = ts.get(ROLLBAR_TS_KEY, undefined)
  return (
    accessToken &&
    new Rollbar({
      accessToken,
      captureUncaught: true,
      captureUnhandledRejections: true,
      payload: {
        environment: es.config.env,
        code_version: es.config.revision,
        source_map_enabled: true
        // person: { // TODO
        //   // id: user && user.user_id,
        //   // username: user && user.username,
        //   // email: user && user.email
        // }
      }
    })
  )
}

export const appAuthIdTokenKey = 'id-token'
export const appAuthAccessTokenKey = 'access-token'
export const appAuthAccessExpiryTokenKey = 'access-token-expiry'

@NgModule({
  imports: [
    HttpClientModule,
    AppRoutingModule,
    NotFoundModule,
    TransferHttpCacheModule,
    SharedModule.forRoot(),
    FirebaseModule,
    Angulartics2Module.forRoot([Angulartics2GoogleAnalytics], {
      pageTracking: {
        clearHash: true
      }
    }),
    BrowserModule.withServerTransition({ appId: 'pm-app' })
    // HttpCacheTagModule.forRoot(
    //   {
    //     provide: CACHE_TAG_CONFIG,
    //     useValue: {
    //       headerKey: 'Cache-Tag',
    //       cacheableResponseCodes: [200]
    //     }
    //   },
    //   {
    //     provide: CACHE_TAG_FACTORY,
    //     useFactory: cacheTagFactory,
    //     deps: [ResponseService]
    //   }
    // )
  ],
  providers: [
    AuthService,
    { provide: AUTH_BEARER_HOSTS, useValue: [] },
    { provide: AUTH_ID_TOKEN_STORAGE_KEY, useValue: appAuthIdTokenKey },
    { provide: AUTH_ACCESS_TOKEN_STORAGE_KEY, useValue: appAuthAccessTokenKey },
    { provide: HTTP_INTERCEPTORS, useClass: HttpAuthInterceptor, multi: true },
    { provide: ErrorHandler, useClass: GlobalErrorHandler },
    { provide: RXJS_DEFAULT_SCHEDULER, useValue: undefined },
    {
      provide: AUTH_ROLES_KEY,
      useFactory: rolesKeyFactory,
      deps: [EnvironmentService]
    },
    {
      provide: AUTH0_CLIENT,
      useFactory: authZeroFactory,
      deps: [EnvironmentService]
    },
    {
      provide: AUTH_ACCESS_TOKEN_EXPIRY_STORAGE_KEY,
      useValue: appAuthAccessExpiryTokenKey
    },
    {
      provide: ROLLBAR_CONFIG,
      useFactory: rollbarFactory,
      deps: [TransferState, EnvironmentService]
    }
  ],
  declarations: [AppComponent],
  bootstrap: [AppComponent],
  exports: [AppComponent]
})
export class AppModule {}
