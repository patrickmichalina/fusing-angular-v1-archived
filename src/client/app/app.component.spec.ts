import { SharedModule } from './shared/shared.module'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { APP_BASE_HREF } from '@angular/common'
import { Route } from '@angular/router'
import { RouterTestingModule } from '@angular/router/testing'
import { Component } from '@angular/core'
import { HomeComponent } from './home/home.component'
import { AppModule } from './app.module'
import { AppBrowserModule } from './app.browser.module'
import { EnvConfig } from '../../../tools/config/app.config'
import { ENV_CONFIG } from './app.config'
import { EnvironmentService } from './shared/services/environment.service'
import { Angulartics2Module } from 'angulartics2'
import { Angulartics2GoogleAnalytics } from 'angulartics2/ga'
import { HttpClientTestingModule } from '@angular/common/http/testing'
import { NavbarService } from './shared/navbar/navbar.service'
import { AppTestingModule } from '../../testing/app-testing.module'

export const TESTING_CONFIG: EnvConfig = {
  siteUrl: 'http://localhost:5000',
  appName: 'fusing-angular',
  // tslint:disable-next-line:max-line-length
  endpoints: {
    api: 'http://localhost:5000/api',
    websocketServer: 'ws://localhost:5001'
  }
}

@Component({
  selector: 'test-cmp',
  template: '<pm-app></pm-app>'
})
class TestComponent {}

describe('App component', () => {
  const config: ReadonlyArray<Route> = [{ path: '', component: HomeComponent }]

  let fixture: ComponentFixture<TestComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [
          AppModule,
          AppTestingModule.forRoot(),
          AppBrowserModule,
          SharedModule,
          HttpClientTestingModule,
          RouterTestingModule.withRoutes(config as any),
          Angulartics2Module.forRoot([Angulartics2GoogleAnalytics])
        ],
        declarations: [TestComponent, HomeComponent],
        providers: [
          { provide: APP_BASE_HREF, useValue: '/' },
          { provide: ENV_CONFIG, useValue: TESTING_CONFIG },
          EnvironmentService,
          NavbarService
        ]
      }).compileComponents()
    })
  )

  beforeEach(
    async(() => {
      fixture = TestBed.createComponent(TestComponent)
    })
  )

  afterEach(
    async(() => {
      TestBed.resetTestingModule()
    })
  )

  it(
    'should build without a problem',
    async(() => {
      expect(fixture.nativeElement).toBeTruthy()
      expect(fixture.nativeElement).toMatchSnapshot()
    })
  )
})
