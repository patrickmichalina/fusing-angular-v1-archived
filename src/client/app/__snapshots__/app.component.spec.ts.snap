// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should build without a problem 1`] = `
<div
  id="root0"
  ng-version="6.0.0-rc.0"
>
  <pm-app>
    <pm-navbar>
      <nav>
        
      </nav>
    </pm-navbar>
    <main>
      <router-outlet />
    </main>
  </pm-app>
</div>
`;
